VultisigApp/VultisigApp/View Models/SendCryptoViewModel.swift

```diff
@@ -149,19 +149,39 @@ class SendCryptoViewModel: ObservableObject, TransferViewModel {
     
     private func getPriceRate(tx: SendTransaction) async -> Double {
         do {
-            let priceRateFiat = await CryptoPriceService.shared.getPrice(priceProviderId: tx.coin.priceProviderId)
-            if priceRateFiat == .zero, SettingsCurrency.current == .USD {
-                let poolInfo = try await CryptoPriceService.shared.fetchCoingeckoPoolPrice(chain: tx.coin.chain, contractAddress: tx.coin.contractAddress)
-                if let priceUsd = poolInfo.price_usd {
-                    return priceUsd
+            
+            var priceRateFiat = Double.zero;
+            
+            if tx.coin.isNativeToken {
+                priceRateFiat = await CryptoPriceService.shared.getPrice(priceProviderId: tx.coin.priceProviderId)
+            } else {
+                // If the price rate is zero and the current currency is USD
+                if tx.coin.chainType == .EVM {
+                    // Attempt to get the token price directly
+                    let tokenPrice = await CryptoPriceService.shared.getTokenPrice(coin: tx.coin)
+                    if tokenPrice != .zero {
+                        return tokenPrice
+                    }
+                    
+                    // Attempt to get the custom token price from CoinGecko
+                    if SettingsCurrency.current == .USD {
```

Comment:
Why is only `.USD`? `https://api.vultisig.com/coingeicko/api/v3/simple/token_price/ethereum?contract_addresses=0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48&vs_currencies=USD,AUD` , it can return others as well

Reply:
Those tokens only have the price if taken from the pools. The suggested endpoint returns {}. So we use the token info in that case.